packages/storage/src/reference.ts

```diff
@@ -132,224 +99,281 @@ export class Reference {
     return this.service;
   }
 
-  /**
-   * Uploads a blob to this object's location.
-   * @param data The blob to upload.
-   * @return An UploadTask that lets you control and
-   *     observe the upload.
-   */
-  put(
-    data: Blob | Uint8Array | ArrayBuffer,
-    metadata: Metadata | null = null
-  ): UploadTask {
-    validate('put', [uploadDataSpec(), metadataSpec(true)], arguments);
-    this.throwIfRoot_('put');
-    return new UploadTask(
-      this,
-      this.service,
-      this.location,
-      this.mappings(),
-      new FbsBlob(data),
-      metadata
-    );
+  get parent(): Reference | null {
+    return getParent(this);
   }
 
-  /**
-   * Uploads a string to this object's location.
-   * @param value The string to upload.
-   * @param format The format of the string to upload.
-   * @return An UploadTask that lets you control and
-   *     observe the upload.
-   */
-  putString(
-    value: string,
-    format: StringFormat = StringFormat.RAW,
-    metadata?: Metadata
-  ): UploadTask {
-    validate(
-      'putString',
-      [stringSpec(), stringSpec(formatValidator, true), metadataSpec(true)],
-      arguments
-    );
-    this.throwIfRoot_('putString');
-    const data = dataFromString(format, value);
-    const metadataClone = Object.assign({}, metadata);
-    if (
-      !type.isDef(metadataClone['contentType']) &&
-      type.isDef(data.contentType)
-    ) {
-      metadataClone['contentType'] = data.contentType!;
+  throwIfRoot_(name: string): void {
+    if (this.location.path === '') {
+      throw invalidRootOperation(name);
     }
-    return new UploadTask(
-      this,
-      this.service,
-      this.location,
-      this.mappings(),
-      new FbsBlob(data.data, true),
-      metadataClone
-    );
   }
+}
 
-  /**
-   * Deletes the object at this location.
-   * @return A promise that resolves if the deletion succeeds.
-   */
-  delete(): Promise<void> {
-    validate('delete', [], arguments);
-    this.throwIfRoot_('delete');
-    return this.service.getAuthToken().then(authToken => {
-      const requestInfo = requests.deleteObject(this.service, this.location);
-      return this.service.makeRequest(requestInfo, authToken).getPromise();
-    });
+/**
+ * Uploads a blob to this object's location.
+ * @public
+ * @param ref - Storage Reference where data should be uploaded.
+ * @param data - The data to upload.
+ * @param metadata - Metadata for the newly uploaded string.
+ * @returns An UploadTask that lets you control and
+ *     observe the upload.
+ */
+export function uploadBytes(
+  ref: Reference,
+  data: Blob | Uint8Array | ArrayBuffer,
+  metadata: Metadata | null = null
+): UploadTask {
+  uploadDataSpec().validator(data);
```

Comment:
I kinda think we should not do this kind of input validation in our new SDKs. This level of validationis pretty uncommon across the JS ecosystem, is a fair bit of work and adds a significant amount of code. I think we should rely on our typings here as much as possible.

This applies everywhere (but we should keep it in the Compat types for backwards compatibility, unless we want to drop it altogether with the next breaking change).

Reply:
Removing validation from both exp and compat per discussion, and following Firestore's example.